pkg/meta/distributed_lock.go

```diff
@@ -0,0 +1,364 @@
+	ctx = logutil.WithCategory(ctx, "distributed-lock")
+	return &DistributedLock{
+		outerCtx:   ctx,
+		store:      store,
+		instanceID: instanceID,
+
+		lockName: []byte(lockName),
+		backoff:  b.backoff,
+		lease:    b.lease,
+	}
+}
+
+func (d *DistributedLock) renewLockLoop(ctx context.Context) {
+	ticker := time.NewTicker(d.lease / 3)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			err := d.renew(ctx)
+			if err != nil {
+				logutil.Logger(d.outerCtx).Warn("renew lock failed", zap.Error(err))
+				return
+			}
+		}
+	}
+}
+
+func (d *DistributedLock) renew(ctx context.Context) error {
+	return d.getLockInTxnWithRetry(ctx,
+		func(lc *lockContent, txn DataTxn) (storeErr, otherErr error) {
+			if lc == nil || lc.InstanceID != d.instanceID {
+				logutil.Logger(d.outerCtx).Warn("lock is unexpectedly removed", zap.String("instanceID", d.instanceID))
+				return nil, nil
+			}
+
+			// Lock is exist, update it.
+			lc.StartTS = txn.StartTS()
+			data, err := lc.encode()
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			err = txn.Set(d.lockName, data)
+			if err != nil {
+				return errors.Trace(err), nil
+			}
+			return nil, nil
+		})
+}
+
+var (
+	// lockHeldErr is returned when the lock is held by another routine.
+	lockHeldErr = errors.Errorf("lock is held")
+	// lockHeldNonBlockingErr is used to indicate that the lock is held and the caller should not retry.
+	lockHeldNonBlockingErr = errors.Errorf("lock is held, non-blocking")
+)
+
+// Lock locks the distributed lock.
+func (d *DistributedLock) Lock() error {
+	err := d.getLockInTxnWithRetry(d.outerCtx,
+		func(lc *lockContent, txn DataTxn) (storeErr, otherErr error) {
+			startTS := txn.StartTS()
+			// Lock is not exist or expired, create or overwrite it.
+			if lc == nil || d.lockExpired(lc, startTS) {
+				lc := newLockContent(startTS, d.instanceID)
+				data, err := lc.encode()
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				err = txn.Set(d.lockName, data)
+				if err != nil {
+					return errors.Trace(err), nil
+				}
+				return nil, nil
+			}
+			// Lock is valid, wait for a while.
+			return lockHeldErr, nil
+		})
+	if err != nil {
+		return err
+	}
+	d.runRenewLoop()
+	return nil
+}
+
+// TryLock tries to lock the distributed lock without blocking.
+func (d *DistributedLock) TryLock() (bool, error) {
+	lockIsHeld := false
+	err := d.getLockInTxnWithRetry(d.outerCtx,
+		func(lc *lockContent, txn DataTxn) (storeErr, otherErr error) {
+			startTS := txn.StartTS()
+			// Lock is not exist or expired, create or overwrite it.
+			if lc == nil || d.lockExpired(lc, startTS) {
+				lc := newLockContent(startTS, d.instanceID)
+				data, err := lc.encode()
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				err = txn.Set(d.lockName, data)
+				if err != nil {
+					return errors.Trace(err), nil
+				}
+				return nil, nil
```

Comment:
Here, the lock should be reset successfully.  
```suggestion
				return ture nil
```